backpack/extensions/firstorder/batch_l2_grad/linear.py

```diff
@@ -1,15 +1,43 @@
+"""Contains batch_l2 extension for Linear."""
 from torch import einsum
 
-from backpack.extensions.firstorder.base import FirstOrderModuleExtension
+from backpack.extensions.firstorder.batch_l2_grad.batch_l2_base import BatchL2Base
 
 
-class BatchL2Linear(FirstOrderModuleExtension):
+class BatchL2Linear(BatchL2Base):
+    """batch_l2 extension for Linear."""
+
     def __init__(self):
+        """Initialization."""
         super().__init__(params=["bias", "weight"])
 
     def bias(self, ext, module, g_inp, g_out, backproped):
+        """batch_l2 for bias.
+
+        Args:
+            ext: backpack extension
+            module: module
+            g_inp: input gradients
+            g_out: output gradients
+            backproped: backpropagation quantities
+
+        Returns:
+            batch_l2 for bias
+        """
         C_axis = 1
         return (g_out[0] ** 2).sum(C_axis)
```

Comment:
This should perform the transformation `S -> J^T S -> (J^T S)**2 -> out`, i.e. the same as the base class. It can thus be deleted. [Here](https://github.com/f-dangel/backpack/blob/development/backpack/core/derivatives/linear.py#L50-L56) is the reference to `J^T`, which is a `no op` if `sum_batch=False`.

Reply:
Deleted.